src/lib.rs

```diff
@@ -622,27 +593,37 @@ encrypted_state_client.as_slice()).unwrap();
                 ("redirect_uri",&self.plugin_config.redirect_uri.as_str()),
                 ("scope", &self.plugin_config.scope),
                 ("claims", &self.plugin_config.claims),
-            ],
-        )
-        .unwrap();
+                ],
+            )
+            .unwrap();
+
+        // Add location header
+        headers.push(("Location", url.as_str()));
 
         // Send HTTP response
         self.send_http_response(
             307,
-            vec![
-                // Original path to redirect back to
-                ("Set-Cookie", &format!("original-path={}; Max-Age={}", &original_path_encoded, 180)),
-                // Set the pkce challenge as a cookie to verify the callback.
-                ("Set-Cookie", &format!("code-verifier={}; Max-Age={}", &pkce_verifier_encoded, 180)),
-                // Set request id as a cookie to verify the callback.
-                ("Set-Cookie", &format!("state={}; Max-Age={}", &encoded_state, 180)),
-                // Set nonce as a cookie to verify the callback.
-                ("Set-Cookie", &format!("nonce={}; Max-Age={}", &encoded_nonce, self.plugin_config.cookie_duration)),
-                // Redirect to `authorization endpoint`
-                ("Location", url.as_str()),
-                ],
-                Some(b"Redirecting..."),
+            // Redirect to `authorization endpoint` along with the cookie
+            headers.to_vec(),
+            Some(b"Redirecting..."),
             );
             return Action::Pause;
     }
+
+    /// Helper function to get the session cookie as a string by getting the cookie from the request
+    /// headers and concatenating all cookie parts.
+    pub fn get_session_cookie_as_string(&self) -> String {
+
+        // Find all cookies that have the cookie_name, split them by ; and remove the name from the cookie
+        // as well as the leading =. Then join the cookie values together again.
+        let cookie = self.get_http_request_header("cookie").unwrap_or_default();
+        // Concatenate all cookie parts
+        let cookie = cookie.split(";")
```

Comment:
The initial cookie for the exchange is shorter and will only have a single part. If there are leftover split parts of the larger cookie, they will be used here and decryption will fail. We should probably store the number of cookie parts in a cookie.